client/app/main/student/student.controller.js

```diff
@@ -2,9 +2,30 @@
 
 var app = angular.module('app');
 
-function StudentCtrl($scope, $state, $stateParams, Student, toastr) {
+function StudentCtrl($scope, $state, $stateParams, Student, 
+  AbsenceRecord, toastr) {
   $scope.student = Student.get({id: $stateParams.id});
 
+  $scope.calcPercentage = function() {
+    AbsenceRecord.current()
+      .$promise.then(function(records) {
+        var totPresent = [];
+        var totEnrolled = [];
+        var addToNext = function(prev, cur) {
+          return prev + cur;
+        };
+        records.forEach(function(record) {
+          totPresent.push(record.entries.present);
+          totEnrolled.push(record.entries.enrolled);
+        });
+        $scope.percentage = Math.floor(
+          (totPresent.reduce(addToNext) / totEnrolled.reduce(addToNext))
+          * 100);
+      });
+  };
+
+  $scope.calcPercentage();
+
```

Comment:
I know there's a better way to do this, but I got stuck trying to navigate the promises and asynch.